### PR TITLE
CO-1633: implement GET /ingredients

### PR DESCRIPTION
**Pull request by jemisonf**

### CODE REVIEW

**Comment by jemisonf** [comment]
Because the logic for processing the filters was _so_ similar to `GET doughs`, I pulled it out into a util module that I'm currently using just for ingredients. I did not do the same for doughs because that would require rewriting the tests there and it seemed out of scope of this ticket, but that would be a definite next step.

File: src/api/v1/db/oracledb/ingredients-dao.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash';
+import { openapi } from 'utils/load-openapi';
+import { serializeIngredients } from 'api/v1/serializers/ingredients-serializer';
+import { GetFilterProcessor } from 'utils/process-get-filters';
+import { getConnection } from './connection';
```
Comment (tsoliangwu0130) [inline src/api/v1/db/oracledb/ingredients-dao.js]: Take a look at our style guide here: https://github.com/osu-mist/style-guides/blob/master/JavaScript.md#import-order

> Whenever possible, the following should be grouped together, separated by a blank line at the top of a module in this order

> Each line within groups 1-3 should be ordered alphabetically by the first alphanumeric character in the name of the package/file in the statement.

File: src/api/v1/db/oracledb/ingredients-dao.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  id: 'ID',
+  ingredientType: 'TYPE',
+  name: 'NAME',
+  notes: 'NOTES',
+};
+
+/**
+ * SQL aliases for mapping INGREDIENTS column names to
+ * IngredientResource property names
+ * @const {string}
```
Comment (tsoliangwu0130) [inline src/api/v1/db/oracledb/ingredients-dao.js]: You probably want a blank line between description and `@const` as you did in dough: https://github.com/osu-mist/pizza-api/blob/master/src/api/v1/db/oracledb/doughs-dao.js#L11

Same comment with your other JSDoc sections.

File: src/utils/process-get-filters.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import _ from 'lodash';
+/**
```
Comment (tsoliangwu0130) [inline src/utils/process-get-filters.js]: Need a blank line between import section and the JSDoc section.

File: src/utils/process-get-filters.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ * Removes the "filter[]" wrapper from a filter parameter
+ *
+ * @param {string} filterName
+ * @returns {string} the filter name with 'filter' and brackets removed
+ */
+const normalizeFilterName = (filterName) => {
+  const filterTextRegex = /filter\[(.*)\]/g;
+  const regexResults = filterTextRegex.exec(filterName);
+  return regexResults ? regexResults[1] : filterName;
```
Comment (tsoliangwu0130) [inline src/utils/process-get-filters.js]: Or even shorter:

```js
return regexResults[1] || filterName;
```

File: src/utils/process-get-filters.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+/**
+ * Returns filters with keys changed to remove the "filter[]" wrapper
+ *
+ * @param {object} filters
+ * @returns {object}
+ */
+const normalizeFilterNames = (filters) => _.mapKeys(filters,
```
Comment (tsoliangwu0130) [inline src/utils/process-get-filters.js]: Not really important, but I guess I'll make `normalizeFilterNames` and `normalizeFilterName` be static methods of `GetFilterProcessor` class.

File: src/utils/process-get-filters.js (hunk 4 above)
Comment (jemisonf) [reply src/utils/process-get-filters.js]: ooh, smart!

File: src/utils/process-get-filters.js (hunk 5 above)
Comment (jemisonf) [reply src/utils/process-get-filters.js]: my goal was to make them private but I think you're right that having them be static is overall the better approach

File: src/utils/process-get-filters.js [hunk 6]
```diff
@@ -1,36 +1,42 @@
...
+    const regexResults = filterTextRegex.exec(filterName);
+    return regexResults[1] || filterName;
+  }
+
   /**
    * Initializes a new filter process using the parameters from the
    * get request for the resource and a map of column names to resource attribute names
-   * @param {array} getParameters
+   *
+   * @param {Array} getParameters
```
Comment (tsoliangwu0130) [inline src/utils/process-get-filters.js]: Array of?

Comment (jemisonf) [reply src/utils/process-get-filters.js]: changed => `Array<object>`, would be handy to have typescript types here :/

Comment (tsoliangwu0130) [reply src/utils/process-get-filters.js]: And it should be `object[]`: https://github.com/osu-mist/students-api/blob/207194b7d9fa346494972ef2fc14773f8e72ee7b/src/utils/jsonapi.js#L20

Comment (jemisonf) [reply src/utils/process-get-filters.js]: this should be actually fixed and the commits should be pushed now

**Pull request merged**